app/scripts/modules/core/src/presentation/tables/Table.tsx

```diff
@@ -0,0 +1,67 @@
+import * as React from 'react';
+
+import { useIsMobile } from 'core/presentation';
+
+import { ITableRowLayoutProps } from './TableRow';
+import { ITableCellLayoutProps } from './TableCell';
+import { TableContext } from './TableContext';
+
+const { useState, useMemo } = React;
+
+export interface ITableColumn {
+  name: string;
+  sortDirection?: 'ascending' | 'descending';
+  onSort?: () => any;
+}
+
+export type ITableColumns = Readonly<Array<string | ITableColumn>>;
+
+export type ITableLayoutProps = Omit<ITableProps, 'layout' | 'columns'> & {
+  columns: ITableColumn[];
+  isMobile: boolean;
+  expandAll: () => any;
+};
+
+export interface ITableLayout {
+  getTableLayout: (props: ITableLayoutProps) => JSX.Element;
+  getTableRowLayout: (props: ITableRowLayoutProps) => JSX.Element;
+  getTableCellLayout: (props: ITableCellLayoutProps) => JSX.Element;
+}
+
+export interface ITableProps {
+  layout: ITableLayout;
+  columns: ITableColumns;
+  expandable?: boolean;
+  children?: React.ReactNode;
+}
+
+export const Table = (props: ITableProps) => {
+  const isMobile = useIsMobile();
+  const [allExpanded, setAllExpanded] = useState(false);
+
+  const { layout, ...propsWithoutLayout } = props;
+
+  const normalizedColumns = props.columns.map(column => (typeof column === 'string' ? { name: column } : column));
+  const tableContext = useMemo(
+    () => ({
+      layout,
+      isMobile,
+      allExpanded,
+      setAllExpanded: () => setAllExpanded(false),
+      expandable: props.expandable,
+      columns: normalizedColumns,
+    }),
+    [layout, isMobile, props.expandable, allExpanded, props.columns],
```

Comment:
yes. I was admittedly kind of lazy here. I think in a theoretical world it'd be more flexible/more performant in some subset of cases to let consumers of these components decide when to invalidate memoization for `columns`, but clearly even the person that wrote it can't be trusted to do that consistently. Given that `columns` isn't a super large/deeply nested object I'm inclined to take over the burden and do what you suggest.

Reply:
Added, thanks! ✅